nats/js/api.py

```diff
@@ -86,6 +87,35 @@ def _to_nanoseconds(val: Optional[float]) -> Optional[int]:
             return 0
         return int(val * _NANOSECOND)
 
+    @staticmethod
+    def _convert_rfc3339(resp: Dict[str, Any], field: str) -> None:
+        """Convert a RFC 3339 formatted string into a datetime.
+
+        If the string is None, None is returned.
+        """
+        val = resp.get(field, None)
+        if val is None:
+            return None
+        raw_date = val[:26]
+        if raw_date.endswith("Z"):
+            raw_date = raw_date[:-1] + "+00:00"
+        resp[field] = datetime.datetime.fromisoformat(raw_date).replace(
+            tzinfo=datetime.timezone.utc
+        )
+
+    @staticmethod
+    def _to_rfc3339(date: datetime.datetime) -> str:
+        """Convert a datetime into RFC 3339 formatted string.
+
+        If datetime does not have timezone information, datetime
+        is assumed to be in UTC timezone.
+        """
+        if date.tzinfo is None:
+            date = date.replace(tzinfo=datetime.timezone.utc)
```

Comment:
I'm not sure about this part. It is described in the PR body though, and there are test cases for it.

I wonder if we should use `.astimezone(datetime.timezone.utc)` instead of `.replace(datetime.timezone.utc)`.

The current implementation considers that date without timezone are UTC, which is the rule that we use at work but may not be universal.

If we change to `astimezone`, the date is assumed to be local and is converted to UTC.